semeio/workflows/correlated_observations_scaling/job.py

```diff
@@ -50,18 +49,15 @@ def scale(self, measured_data):
         matrix = DataMatrix(measured_data.data)
         matrix.normalize_by_std()
 
-        scale_factor = matrix.get_scaling_factor(config.CALCULATE_KEYS)
         events = config.CALCULATE_KEYS
-        data_matrix = matrix.get_data_matrix()
         nr_components, singular_values = matrix.get_nr_primary_components(
```

Comment:
This is more of a question than anything else. Does it make sense to have the `DataMatrix` instance store the `nr_components`? And in that case it is not necessary to keep something that is strictly related to the instance and feed it back again. If it wasn't for the `self._reporter.publish` of `singular_values` I would think that the `get_scaling_factor` could've checked if the `nr_components` was calculated, and if not, calculate them..

Not a big concern though, this PR already improves the code flow.

Reply:
Dont think that makes sense, because then the `DataMatrix`is more locked to doing only this kind of scaling. I think it makes more sense then to move the `get_scaling_factor` to a stand-alone function in that case.

Reply:
Turns out it was already logged on the other side.

Reply:
Let's not include it to the `DataMatrix` at least then.